src/sections/collection/collection-items-panel/CollectionItemsPanel.tsx

```diff
@@ -0,0 +1,225 @@
+import { useEffect, useRef, useState } from 'react'
+import { useSearchParams } from 'react-router-dom'
+import { CollectionRepository } from '@/collection/domain/repositories/CollectionRepository'
+import { CollectionItemsPaginationInfo } from '@/collection/domain/models/CollectionItemsPaginationInfo'
+import { CollectionSearchCriteria } from '@/collection/domain/models/CollectionSearchCriteria'
+import { CollectionItemType } from '@/collection/domain/models/CollectionItemType'
+import { useGetAccumulatedItems } from './useGetAccumulatedItems'
+import { UseCollectionQueryParamsReturnType } from '../useGetCollectionQueryParams'
+import { useLoadMoreOnPopStateEvent } from './useLoadMoreOnPopStateEvent'
+import { useLoading } from '@/sections/loading/LoadingContext'
+import { QueryParamKey } from '../../Route.enum'
+import { CollectionHelper } from '../CollectionHelper'
+import { FilterPanel } from './filter-panel/FilterPanel'
+import { ItemsList } from './items-list/ItemsList'
+import { SearchPanel } from './search-panel/SearchPanel'
+import { ItemTypeChange } from './filter-panel/type-filters/TypeFilters'
+import styles from './CollectionItemsPanel.module.scss'
+
+interface CollectionItemsPanelProps {
+  collectionId: string
+  collectionRepository: CollectionRepository
+  collectionQueryParams: UseCollectionQueryParamsReturnType
+  addDataSlot: JSX.Element | null
+}
+
+/**
+ * HOW IT WORKS:
+ * This component loads items on 5 different scenarios:
+ * 1. When the component mounts
+ * 2. When the user scrolls to the bottom of the list and there are more items to load
+ * 3. When the user submits a search query in the search panel
+ * 4. When the user changes the item types in the filter panel
+ * 5. When the user navigates back and forward in the browser
+ *
+ * It initializes the search criteria with the query params in the URL.
+ * By default if no query params are present in the URL, the search query is empty and the item types are COLLECTION and DATASET.
+ * Every time a load of items is triggered, the pagination info is updated and the URL is updated with the new query params so it can be shared and the user can navigate back and forward in the browser.
+ */
+
```

Comment:
In testing the search, I noticed that the JSF version is highlighting the search text in the results, but we aren't doing that in the SPA. Should we make a new issue for that?

<img width="1087" alt="Screenshot 2024-09-30 at 4 51 22 PM" src="https://github.com/user-attachments/assets/4559c019-1dc0-4826-b1b5-f055f8d9905c">

JSF Version: image is highlighted in title and file details
<img width="973" alt="Screenshot 2024-09-30 at 4 50 49 PM" src="https://github.com/user-attachments/assets/2ae3445b-3f92-4f9a-ad20-534a358e0928">

Reply:
Yes could be nice to have it, could be a  good use maybe of the [html mark element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/mark)